epam_python_course/L2/HW/task2_select_filter_dict.py

```diff
@@ -0,0 +1,137 @@
+"""
+Данные:
+
+friends = [
+    {'name': 'Сэм', 'gender': 'Мужской', 'sport': 'Баскетбол', 'email': 'email@email.com'},
+    {'name': 'Эмили', 'gender': 'Женский', 'sport': 'Волейбол', 'email': 'email1@email1.com'},
+    ...
+]
+
+Необходимо создать функции:
+1. select(*field_name: list) - функция принимает список полей, которые должны быть в результирующем списке
+2. field_filter(field_name: str, *collection: list) - функция принимает имя по которому необходимо фильтровать и
+итерируемый объект - значения которые должны быть в результирующем списке
+3. query(*collection: list, select: function, field_filter: function, ...) -> list - функция которая действительно
+выбирает нужные поля и фильтрует их. Выбор значений и фильтрация предполагается делать один раз(!), т.е. сначала нужно
+вычислить все поля а затем реально работать с данными.
+
+Пример работы программы:
+result = query(
+    friends,
+    select('name', 'gender', 'sport'),
+    field_filter('sport', ['Баскетбол', 'Волейбол']),
+    field_filter('gender', ['Мужской']),
+)
+print(result) # [{'name': 'Сэм', 'gender': 'Мужской', 'sport': 'Баскетбол'}, ]
+
+Дополнительно
+1. Красивый, читаемый код
+2. Документирование функций обязательно. Используем стиль документации Sphinx
+"""
+
+
+def select(*field_name: list):
+    """
+    Select required field from record and return updated version only with required fields
+
+    :param field_name: required field for selection
+    :type field_name: list
+    :return: updated record
+    :rtype: dict
+    """
+
+    def inner_select(record: dict):
+        # Inner selection work with record and return new instance only with required fields
+        updated_record = {}
+        for key in field_name:
+            updated_record[key] = record[key]
+        return updated_record
+
+    return inner_select
+
+
+def field_filter(field_name: str, collection: list):
```

Comment:
look at https://docs.python.org/3/library/functions.html#filter
maybe it gives you some thoughts =)

Reply:
I think in lecture #2 suggested implement filter without iterator/generator. But after lecture #7  I should will implement already with iterator/generator. 
Thank for idea! I didn't check built-in function in this tasks, but it's useful to check similar functions for more common approach